pkg/controller/contrailmonitor/contrailmonitor_controller.go

```diff
@@ -0,0 +1,453 @@
+		serIns4.Status = "NotActive"
+	} else {
+		serIns4.Status = "Active"
+	}
+	_, err = controllerutil.CreateOrUpdate(context.Background(), r.client, serIns4, func() error {
+		return controllerutil.SetControllerReference(instance, serIns4, r.scheme)
+	})
+	if err != nil {
+		return reconcile.Result{}, nil
+	}
+
+	wklist, err := r.getWebuilist()
+	fmt.Println(wklist.Items[0].Status.ServiceStatus)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	wcount := len(wklist.Items)
+	if wcount > 0 {
+		for i := 0; i < wcount; i++ {
+			for _, value := range wklist.Items[i].Status.ServiceStatus {
+				for _, n := range value {
+					serIns8 := &contrailv1alpha1.Contrailstatusmonitor{ObjectMeta: metav1.ObjectMeta{Name: n.ModuleName, Namespace: "contrail"}}
+
+					serIns8.Status = n.ModuleState
+					_, err := controllerutil.CreateOrUpdate(context.Background(), r.client, serIns8, func() error {
+						return controllerutil.SetControllerReference(instance, serIns8, r.scheme)
+					})
+					if err != nil {
+						return reconcile.Result{}, err
+					}
+				}
+			}
+		}
+	} else {
+
+		serIns8 := &contrailv1alpha1.Contrailstatusmonitor{ObjectMeta: metav1.ObjectMeta{Name: "Webui", Namespace: "contrail"}}
+
+		serIns8.Status = "NotActive"
+		_, err := controllerutil.CreateOrUpdate(context.Background(), r.client, serIns8, func() error {
+			return controllerutil.SetControllerReference(instance, serIns8, r.scheme)
+		})
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+	}
+
+	clist, err := r.getConfiglist()
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	fmt.Println(clist.Items[0].Status.ServiceStatus)
+	ccount := len(clist.Items)
+	if ccount > 0 {
+		for j := 0; j < ccount; j++ {
+			for _, value := range clist.Items[j].Status.ServiceStatus {
+				for _, n := range value {
+					serIns7 := &contrailv1alpha1.Contrailstatusmonitor{ObjectMeta: metav1.ObjectMeta{Name: n.ModuleName, Namespace: "contrail"}}
+
+					serIns7.Status = n.ModuleState
+					_, err := controllerutil.CreateOrUpdate(context.Background(), r.client, serIns7, func() error {
+						return controllerutil.SetControllerReference(instance, serIns7, r.scheme)
+					})
+					if err != nil {
+						return reconcile.Result{}, err
+					}
+				}
+			}
+		}
+	} else {
+		serIns7 := &contrailv1alpha1.Contrailstatusmonitor{ObjectMeta: metav1.ObjectMeta{Name: "config", Namespace: "contrail"}}
+
+		serIns7.Status = "NotActive"
+		_, err := controllerutil.CreateOrUpdate(context.Background(), r.client, serIns7, func() error {
+			return controllerutil.SetControllerReference(instance, serIns7, r.scheme)
+		})
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+	}
+	provisionmanager, err := r.getProvisionmanager(instance)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	serIns3 := &contrailv1alpha1.Contrailstatusmonitor{ObjectMeta: metav1.ObjectMeta{Name: provisionmanager.Name, Namespace: "contrail"}}
+
+	if provisionmanager.Status.Active == nil {
```

Comment:
What if status.Active is false?

Reply:
Will we report "Active"?

Reply:
Yes, WE need to report Active/NotActive for postgres, provisionmanager, zookeeper, cassandra etc.

Yes, I have noticed  the point that you have raised, These fields in the crds are boolean pointers, I can check boolean pointer as  the GCP build raised Panic error,  which is not raised by the blazel build, so, after couple of iterations , I have decided with 2 values nil and true, I tried to even check the IsAlive function that cassandra, zookeper, rabbitmq used to to test the pod activity, I noticed they are very inconsiatnat in returning the value.